### PR TITLE
Update dallas.rst

### PR DESCRIPTION
**Pull request by frspp**
Added: line 41 note from io

## Description:

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<esphome PR number goes here>

## Checklist:

  - [ ] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by glmnet** [comment]
Did you manage to setup this with an input only pin?

**Comment by frspp** [comment]
> Did you manage to setup this with an input only pin?

No. It needs input and output.

**Comment by glmnet** [comment]
Ahh, my poor English.

I mean if you tried to get this running with an input only pin and failed of course, in that case how you realized it wasn't possible. I did not check, but ESPHome should tell you in this case the configuration is invalid

**Comment by frspp** [comment]
> Ahh, my poor English.
> 
> I mean if you tried to get this running with an input only pin and failed of course, in that case how you realized it wasn't possible. I did not check, but ESPHome should tell you in this case the configuration is invalid

Sensor did not work so I did some googling and read that it needs IO. Compile/logger did not give anything about the sensor. Changed to another pin and still the sensor is not working so there is something wrong with my setup still.

**Comment by glmnet** [comment]
which pin you've tried initially?

**Comment by glmnet** [comment]
I'm adding this check https://github.com/esphome/esphome/pull/1692 so this should be catch by validation.

**Comment by frspp** [comment]
> which pin you've tried initially?

Board is NodeMCU ESP32 38pin. First I tried pin GPIO39 ('SN'). 

Now it is on GPIO32, which is closest IO pin to 3v3.

File: components/sensor/dallas.rst [hunk 1]
```diff
@@ -20,8 +20,8 @@ and similar 1-Wire temperature sensors.
 To use your :ref:`dallas sensor <dallas-sensor>`, first define a dallas “hub” with a pin and
 id, which you will later use to create the sensors. The 1-Wire bus the
 sensors are connected to should have an external pullup resistor of
-about 4.7KΩ. For this, connect a resistor of *about* 4.7KΩ (values around that like 1KΩ will, if you don't have
-massively long wires, work fine in most cases) between ``3.3V`` and the data pin.
+about 4.7KΩ. For this, connect a resistor of *about* 4.7KΩ (values around that like 1KΩ will work fine, if you don't have
+massively long wires, in most cases) between ``3.3V`` and the data pin.
```
Comment (frenck) [inline components/sensor/dallas.rst]:
```suggestion
about 4.7KΩ. For this, connect a resistor of *about* 4.7KΩ between ``3.3V``
and the data pin. Values +/- 1KΩ will, in most cases, work fine as well,
if you don't have massively long wires.
```

File: components/sensor/dallas.rst [hunk 2]
```diff
@@ -38,7 +38,7 @@ massively long wires, work fine in most cases) between ``3.3V`` and the data pin
 Configuration variables:
 ************************
 
-- **pin** (**Required**, number): The pin the sensor bus is connected to.
+- **pin** (**Required**, number): The pin the sensor bus is connected to. NOTE: 1-wire is a bi-directional bus so it requires both input and output from pin.
```
Comment (frenck) [inline components/sensor/dallas.rst]:
```suggestion
- **pin** (**Required**, number): The pin the sensor bus is connected to. Please note that 1-wire is a bi-directional bus so it requires both input and output from the pin.
```

**Review by frenck — CHANGES_REQUESTED** [review]
Thanks, @frspp!

Left some small rewording suggestions. Could you take a look?

Thanks! 👍 

../Frenck 🚂

File: components/sensor/dallas.rst [hunk 3]
```diff
@@ -20,8 +20,9 @@ and similar 1-Wire temperature sensors.
 To use your :ref:`dallas sensor <dallas-sensor>`, first define a dallas “hub” with a pin and
 id, which you will later use to create the sensors. The 1-Wire bus the
 sensors are connected to should have an external pullup resistor of
-about 4.7KΩ. For this, connect a resistor of *about* 4.7KΩ (values around that like 1KΩ will, if you don't have
-massively long wires, work fine in most cases) between ``3.3V`` and the data pin.
+about 4.7KΩ. For this, connect a resistor of *about* 4.7KΩ between ``3.3V``
+and the data pin. Values +/- 1KΩ will, in most cases, work fine as well,
```
Comment (oxan) [inline components/sensor/dallas.rst]:
```suggestion
and the data pin. Values ± 1KΩ will, in most cases, work fine as well,
```

**Review by oxan — CHANGES_REQUESTED** [review]

**Pull request merged**